tests/engine/test_generation.py

```diff
@@ -38,6 +39,7 @@ def test_empty_bucket():
 
 
 def test_empty_bucket_two():
+    """Test the generation of two empty buckets."""
```

Comment:
One of the buckets isn't empty. It's an other test to ensure empty buckets can be created in the middle of a file